.github/workflows/check_labels.yml

```diff
@@ -5,44 +5,88 @@ on:
     branches-ignore:
       - 'main'
+      - 'release/**'
     types: [ synchronize, labeled, unlabeled ]
 
 concurrency:
   group: ${{ github.workflow }}-${{ github.head_ref }}
   cancel-in-progress: true
 
 jobs:
+  get_allowed_labels:
+    runs-on: ubuntu-latest
+
+    outputs:
+      allowed-labels: ${{ steps.get_allowed_labels.outputs.allowed_labels }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Get the list of allowed pull request labels
+        id: get_allowed_labels
+        env:
+          PROJECT_ROOT: ${{ github.workspace }}
+        run: |
+          RED='\033[0;31m'
+          FILE_NAME=.release_notes_allowed_labels_list
+          GITHUB_DIR=.github
+          FILE_PATH=$PROJECT_ROOT/$GITHUB_DIR/$FILE_NAME
+          if [[ ! -f "$FILE_PATH" ]]; then
+             echo -e "${RED}$FILE_NAME file doesn't exits in $GITHUB_DIR/"
+             exit 1
+          fi 
+          ALLOWED_LABELS=$(cat $FILE_PATH)
+          echo -e "allowed_labels=$ALLOWED_LABELS" >> $GITHUB_OUTPUT
+          echo -e "Allowed labels are: $ALLOWED_LABELS"
+
   labels-check:
-    # https://github.com/actions/virtual-environments/
     runs-on: ubuntu-latest
+    needs: get_allowed_labels
 
     steps:
       - uses: actions/checkout@v4
-      - name: Check PR labels
+
+      - name: Validate release notes
+        env:
+          REQUIRED_PR_LABELS: "Breaking change,Feature,Fix"
+          OPTIONAL_PR_LABELS: "Dependencies,Chore"
+          ALLOWED_RELEASE_NOTES_LABELS: ${{ needs.get_allowed_labels.outputs.allowed-labels }}
+          PR_BODY: ${{ github.event.pull_request.body }}
         run: |
-          all_pr_labels_json=$(cat <<EOF
+          # Fetch PR labels
+          ALL_PR_LABELS_JSON=$(cat <<EOF
           ${{ toJson(github.event.pull_request.labels.*.name) }}
           EOF
           )
-          all_pr_labels=$(jq -r '.[]' <<< "$all_pr_labels_json")
-          echo "PR labels: $all_pr_labels"
-
-          if [[ "${all_pr_labels[@]}" =~ 'Breaking change' || "${all_pr_labels[@]}" =~ 'Feature' || "${all_pr_labels[@]}" =~ 'Fix' ]]
-          then
-            echo "Checking if release notes were added..."
-            git fetch origin develop --depth 1
-            if [ -n "$(git diff origin/develop RELEASE_NOTES.md)" ]
-            then
-              echo "Release notes were updated."
-              exit 0
-            else
-              echo "::error::Add release notes for your PR by updating RELEASE_NOTES.md"
-              exit 1
+          ALL_PR_LABELS=$(jq -r '.[]' <<< "$ALL_PR_LABELS_JSON")
+          
+          # Read PR labels into an array
+          IFS=',' read -r -a PR_LABELS <<< "$ALL_PR_LABELS"
+          echo "PR labels: $PR_LABELS"
+
+          # Initialize flags
+          REQUIRED_LABEL_MATCH=false
+          OPTIONAL_LABEL_MATCH=false
+
+          # Check labels against categories
+          for label in "${PR_LABELS[@]}"; do
+            if [[ ",$REQUIRED_PR_LABELS," == *",$label,"* ]]; then
```

Comment:
There probably isn't a cleaner way to find an item in a list, right?

Reply:
I couldn't find a better way when implementing, since this checks if a label exists in a comma-separated list. Otherwise I guess we could create another list of required and optional pr labels and run two loops, but in my opinion it is easier to read this way.

Reply:
I added comments for more clarity.